homeassistant/components/redgtech/device.py

```diff
@@ -0,0 +1,32 @@
+"""Device representation for Redgtech integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+
+@dataclass
```

Comment:
This is not a good fit for a dataclass IMO considering the custom `__init__` method
I'd suggest to remove the custom `__init__` and let the coordinator pass in the values directly instead.